pkg/networkservice/common/connect/server.go

```diff
@@ -34,22 +36,39 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
+// Contain grpc connection and client connection associated
+type clientEntry struct {
+	clientURI       *url.URL
+	client          networkservice.NetworkServiceClient
+	cc              grpc.ClientConnInterface
+	ready           chan bool // Flag for connection is establishing
+	connections     map[string]*networkservice.Connection
+	err             error
+	closeConnection func() error
+}
+
+func (ce *clientEntry) markAsReady(state bool) {
+	ce.ready <- state
+	close(ce.ready)
+}
+
 type connectServer struct {
-	ctx                     context.Context
-	dialOptions             []grpc.DialOption
-	clientFactory           func(ctx context.Context, conn grpc.ClientConnInterface) networkservice.NetworkServiceClient
-	uRLToClientMap          clientMap // key == url as string
-	connectionIDToClientMap clientMap // key == connection.GetId()
-	dialOptionFactories     []func(ctx context.Context, request *networkservice.NetworkServiceRequest, clientURL *url.URL) []grpc.DialOption
+	ctx                 context.Context
+	dialOptions         []grpc.DialOption
+	clientFactory       func(ctx context.Context, conn grpc.ClientConnInterface) networkservice.NetworkServiceClient
+	clientMap           map[string]*clientEntry // key == url as string
+	connections         map[string]*clientEntry // Connection map is required to close using connection id.
```

Comment:
'clientMap' and 'connections' are both better names :)
Would it make sense for 'clientMap' -> 'clients' to match 'connections' ?

Reply:
Sounds good.